### PR TITLE
[NUI] Enhance Text documents

### PR DESCRIPTION
**Pull request by wonrst**
Signed-off-by: Bowon Ryu <bowon.ryu@samsung.com>

### Change Description ###
Some descriptions of the properties in NUI Text components are insufficient.
This commit enhances the document.

### CODE REVIEW

File: docs/application/dotnet/guides/nui/text.md [hunk 1]
```diff
@@ -396,12 +412,28 @@ field.PlaceholderTextFocused = "Enter Name";
...
 ```
 
-When the `TextField` is tapped, it automatically gets the keyboard focus. Key events correspond to entering the text. Additionally, the placeholder text is removed as soon as the text is entered. The text entered can be retrieved by using the `TEXT` property:
+When the `TextField` is tapped, it automatically gets the keyboard focus. Key events correspond to entering the text. Additionally, the placeholder text is removed as soon as the text is entered. The text entered can be retrieved by using the `Text` property:
 
 ```csharp
 string fieldTextString = field.Text;
 ```
 
+### Use Placeholder Property in TextField
```
Comment (nawab-samsung) [inline docs/application/dotnet/guides/nui/text.md:421]: Global: 
Can we use the Sentence case for all headings? Let me know if that is possible.

File: docs/application/dotnet/guides/nui/text.md [hunk 2]
```diff
@@ -432,6 +487,34 @@ field.TextChanged += (obj, e) => {
 };
 ```
 
+### Decoration of Grab Handle and Selection Handle
+
+To decorate the grab handle and selection handle, you can use the `GrabHandleImage`, `SelectionHandleImageLeft`, `SelectionHandleImageRight` properties.
```
Comment (nawab-samsung) [inline docs/application/dotnet/guides/nui/text.md:492]:
```suggestion
To decorate the grab handle and selection handle, you can use the `GrabHandleImage`, `SelectionHandleImageLeft`, and `SelectionHandleImageRight` properties.

File: docs/application/dotnet/guides/nui/text.md [hunk 3]
```diff
@@ -432,6 +487,34 @@ field.TextChanged += (obj, e) => {
 };
 ```
 
+### Decoration of Grab Handle and Selection Handle
+
+To decorate the grab handle and selection handle, you can use the `GrabHandleImage`, `SelectionHandleImageLeft`, `SelectionHandleImageRight` properties.
+
+Also you can change the color of handle using `GrabHandleColor` property.
```
Comment (nawab-samsung) [inline docs/application/dotnet/guides/nui/text.md:494]:
```suggestion
Also, you can change the color of the handle using the `GrabHandleColor` property.

**Review by nawab-samsung — CHANGES_REQUESTED** [review]
@wonrst Please incorporate the comments.

File: docs/application/dotnet/guides/nui/text.md (hunk 1 above)
Comment (wonrst) [reply docs/application/dotnet/guides/nui/text.md:421]: I am sorry, what is Sentence case?

File: docs/application/dotnet/guides/nui/text.md [hunk 4]
```diff
@@ -225,6 +225,22 @@ To use the text-decoration, set the applicable property:
 
     ![Text with color underline](media/TextWithColorUnderline.png)
 
+-   To outline the text label, set the `Outline` property::
```
[the remark this answers is not in the document]
Comment (wonrst) [reply docs/application/dotnet/guides/nui/text.md]: Done

File: docs/application/dotnet/guides/nui/text.md (hunk 2 above)
Comment (wonrst) [reply docs/application/dotnet/guides/nui/text.md]: Done

File: docs/application/dotnet/guides/nui/text.md (hunk 3 above)
Comment (wonrst) [reply docs/application/dotnet/guides/nui/text.md]: Done

File: docs/application/dotnet/guides/nui/text.md (hunk 1 above)
Comment (wonrst) [reply docs/application/dotnet/guides/nui/text.md:421]: Class or property names cannot be written in lowercase.
In other cases I can change it.

For example :
### Use Placeholder Property in TextField -> ### Use Placeholder property in TextField.
(Placeholder = property, TextField = class)

### Align Text in TextField -> ### Align text in TextField
### Use HiddenInputSettings Property in TextField -> ### Use HiddenInputSettings property in TextField

Please let me know if you want this change for all headings.

Comment (nawab-samsung) [reply docs/application/dotnet/guides/nui/text.md:421]: Oh sorry! I didn't mean that exactly.

What my intention was that heading of this page which starts with (#, ##, ###) are, written in Title case (we capitalize ist letter of each word in a title or heading)

I suggested using sentence case for headings, which means:

In sentence case, lowercase most words in a title or heading. Capitalize only the Ist letter of the sentence: 


e.g 

### Use Placeholder Property  (Title case)

### Use placeholder property (Sentence case) 

If there is a name in the sentence case, we write that as capital.

### Use placeholder property in TextField (correct)
### Use placeholder property in textfield (Incorrect)

**Review by nawab-samsung — CHANGES_REQUESTED** [review]
@wonrst 
I have elaborated on the doubt. Please apply the comment. 
Thanks.

File: docs/application/dotnet/guides/nui/text.md (hunk 1 above)
Comment (wonrst) [reply docs/application/dotnet/guides/nui/text.md:421]: Thanks for your explanation.

By the way, in below case,
### Use placeholder property in TextField (correct)


"Placeholder" is the name of the property(attribute).
So, I think below is correct.
What do you think?
### Use Placeholder property in TextField

**Comment by nawab-samsung** [comment]
@wonrst Gentle reminder, Please respond to the comments. 
Thanks.

**Comment by wonrst** [comment]
@nawab-samsung 
Hello, Please check this :)

> Thanks for your explanation.
> 
> By the way, in below case,
> 
> ### Use placeholder property in TextField (correct)
> "Placeholder" is the name of the property(attribute).
> So, I think below is correct.
> What do you think?
> 
> ### Use Placeholder property in TextField

**Comment by nawab-samsung** [comment]
@wonrst Yes, you got it correctly.

**Comment by wonrst** [comment]
@nawab-samsung 
Thanks, I used sentence case for all headings.

File: docs/application/dotnet/guides/nui/text.md [hunk 5]
```diff
@@ -617,7 +617,7 @@ editor.Text = "This is a multiline text.\n I can write several lines.\n";
 Window.Instance.Add(editor);
 ```
 
-### TextEditor Properties
+### TextEditor properties
```
Comment (nawab-samsung) [inline docs/application/dotnet/guides/nui/text.md:620]: Line 682 
## Related information

File: docs/application/dotnet/guides/nui/text.md [hunk 6]
```diff
@@ -417,7 +449,30 @@ The following example illustrates text alignment:
 field.HorizontalAlignment = HorizontalAlignment.Begin;
 ```
 
-### Use Input Properties in TextField
+### Use HiddenInputSettings property in TextField
+
+The `TextField` class can substitute the input text using the `HiddenInputSettings` property.
+
+To set the `HiddenInputSettings` for detail options, you can use [HiddenInputProperty](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.HiddenInputProperty.html) and [HiddenInputModeType](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.HiddenInputModeType.html).
```
Comment (nawab-samsung) [inline docs/application/dotnet/guides/nui/text.md:456]: Just one doubt :  To set the `HiddenInputSettings` for detailed options, 
What are the detailed options? 

I am not able to get this, can you elaborate. 
```suggestion
To set the `HiddenInputSettings` for detailed options, you can use [HiddenInputProperty](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.HiddenInputProperty.html) and [HiddenInputModeType](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.HiddenInputModeType.html).

**Review by nawab-samsung — CHANGES_REQUESTED** [review]
@wonrst 
Thanks for changing the headings, I have a doubt, request you to kindly have a look.

File: docs/application/dotnet/guides/nui/text.md (hunk 5 above)
Comment (wonrst) [reply docs/application/dotnet/guides/nui/text.md:620]: Done

File: docs/application/dotnet/guides/nui/text.md (hunk 6 above)
Comment (wonrst) [reply docs/application/dotnet/guides/nui/text.md]: @nawab-samsung 

The detailed options are Mode and Properties of HiddenInputSettings.
(Struct HiddenInputProperty, Enum HiddenInputModeType)

Depending on HiddenInputModeType, TextField behaves as follows.
* HideAll : Hide all the input text.
* HideCount : Hide n characters from start.
* HideNone : Don't hide text.
* ShowCount : Show n characters from start.
* ShowLastCharacter : Show last character for the duration(use ShowLastCharacterDuration property to modify duration).

Also, depending on the mode, through HiddenInputProperty,
ShowLastCharacterDuration, SubstituteCharacter, SubstituteCount values ​​can be set.

I think, users need a detailed explanation of this.
So I made an API reference link for this. (The most detailed explanation)

Comment (nawab-samsung) [reply docs/application/dotnet/guides/nui/text.md]: @wonrst thanks.

**Comment by wonrst** [comment]
@nawab-samsung Hello,
I am not authorized to merge this pull request.

**Review by annie-samsung — COMMENTED** [review]
@wonrst, there is so much content in this page that I think readability is affected. My suggestion would be to rearrange the content either like code and example image side by side within a table.

OR to break this into three separate topics: 
TextField
TextLabel
TextEditor

These are just my opinion, probably you can consider this for a separate PR.
@mijin85cho thoughts?

**Comment by mijin85cho** [comment]
> @wonrst, there is so much content in this page that I think readability is affected. My suggestion would be to rearrange the content either like code and example image side by side within a table.
> 
> OR to break this into three separate topics:
> TextField
> TextLabel
> TextEditor
> 
> These are just my opinion, probably you can consider this for a separate PR.
> @mijin85cho thoughts?

In my opinion, 
in this page the toc menu on the right and the links in the starting point are provided to make it easy to go to the location where a user want to read. So, as a web page, I think current structure is also acceptable.

image

Also, I'm worrying the https://github.com/Samsung/tizen-docs/pull/1387. In this PR dotnet/guides/nui/text.md is also updated as you can see. Isn't this PR in dotnet_ui_2.0 branch must be merged with master? If there are many differences between two branches, there would be lots of conflicts that must be resolved. I'm worrying if you can handle that. 

**FYI :** There were 3 pages for TextLabel, TextField, and TextEditor before, and they were merged by https://github.com/Samsung/tizen-docs/pull/1009.

**Comment by nawab-samsung** [comment]
@wonrst Can you please have a look at the comments and provide your suggestions.

File: docs/application/dotnet/guides/nui/text.md [hunk 7]
```diff
@@ -225,6 +225,22 @@ To use the text-decoration, set the applicable property:
...
+    TextLabel label = new TextLabel("Text with Color Outline");
+    label.TextColor = Color.White;
+    PropertyMap outline = new PropertyMap();
+    outline.Add("color", new PropertyValue(Color.Red));
+    outline.Add("width", new PropertyValue(2.0f)); /// 2-pixel width
+    label.Outline = outline;
+    Window.Instance.Add(label);
+    ```
+
+    **Figure: Text with red outline**
```
[the remark this answers is not in the document]
Comment (wonrst) [reply docs/application/dotnet/guides/nui/text.md:240]: Hi @annie-samsung ,

If I add more white space, will there be more space?
I don't think so.

If there is 1 white space or 2 more spaces, the result looks the same.
Could you please tell me how? then I will update it.

**Comment by wonrst** [comment]
Hello @mijin85cho ,

I have no idea about dotnet_ui_2.0.
Anyway, if necessary I can update the same content.
But how about making another PR?
I can check its history from @JoogabYun

File: docs/application/dotnet/guides/nui/text.md [hunk 8]
```diff
@@ -596,7 +679,7 @@ The following table lists the available `TextEditor` properties:
 | `TranslatableText`                 | String      | Specifies the `TranslatableText` property that sets the SID value. |
 
 
-## Related Information
+## Related information
 - Examples
   - [TvTextSample](https://github.com/Samsung/Tizen-CSharp-Samples/tree/master/TV/TextSample)
```
[the remark this answers is not in the document]
Comment (wonrst) [reply docs/application/dotnet/guides/nui/text.md]: Done

**Comment by annie-samsung** [comment]
> > @wonrst, there is so much content in this page that I think readability is affected. My suggestion would be to rearrange the content either like code and example image side by side within a table.
> > OR to break this into three separate topics:
> > TextField
> > TextLabel
> > TextEditor
> > These are just my opinion, probably you can consider this for a separate PR.
> > @mijin85cho thoughts?
> 
> In my opinion,
> in this page the toc menu on the right and the links in the starting point are provided to make it easy to go to the location where a user want to read. So, as a web page, I think current structure is also acceptable.
> 
> image
> 
> Also, I'm worrying the #1387. In this PR dotnet/guides/nui/text.md is also updated as you can see. Isn't this PR in dotnet_ui_2.0 branch must be merged with master? If there are many differences between two branches, there would be lots of conflicts that must be resolved. I'm worrying if you can handle that.
> 
> **FYI :** There were 3 pages for TextLabel, TextField, and TextEditor before, and they were merged by #1009.

@wonrst, @mijin85cho, thanks for your opinion. I think for now we can leave it as is.
It is evident this page is created by emerging 3 topics in an as is manner hence not so organized. Will wait for 1387 to be merged first.

File: docs/application/dotnet/guides/nui/text.md [hunk 9]
```diff
@@ -417,7 +449,30 @@ The following example illustrates text alignment:
 field.HorizontalAlignment = HorizontalAlignment.Begin;
 ```
 
-### Use Input Properties in TextField
+### Use HiddenInputSettings property in TextField
+
+The `TextField` class can substitute the input text using the `HiddenInputSettings` property.
+
+To set the `HiddenInputSettings` for detailed options, you can use [HiddenInputProperty](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.HiddenInputProperty.html) and [HiddenInputModeType](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.HiddenInputModeType.html).
```
Comment (mijin85cho) [inline docs/application/dotnet/guides/nui/text.md:456]: Hello, @annie-samsung 

As I commented on messenger, would you please check the links to API here?

.NET guide are updated by https://github.com/Samsung/tizen-docs/pull/1445,
and one of the update is changing the links to API.

So I think these links should be checked with author of https://github.com/Samsung/tizen-docs/pull/1445
whether if it is ok to use https://samsung.github.io/ link.

Or, another temporary solution in my thought is,
removing the links to `HiddenInputProperty` and `HiddenInputModeType` for now, and adding it back later.


Would you please check?

Comment (annie-samsung) [reply docs/application/dotnet/guides/nui/text.md:456]: @mijin85cho , sure will sort this out

Comment (annie-samsung) [reply docs/application/dotnet/guides/nui/text.md:456]: @piotrekgorny, in this PR you can see that in text.md file at line 456 github.io links are provided, could you suggest how do we go about with this PR, do we remove them, an alternate link is available?

@wonrst, Piotr's team just finished a big exercise of removing github.io links from all NUI guides in PR
https://github.com/Samsung/tizen-docs/pull/1387, so adding it back via this PR can cause problem.

Comment (annie-samsung) [inline docs/application/dotnet/guides/nui/text.md:456]:
```suggestion
To set the `HiddenInputSettings` for detailed options, you can use [HiddenInputProperty](/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.HiddenInputProperty.html) and [HiddenInputModeType](/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.HiddenInputModeType.html).

```

**Comment by annie-samsung** [comment]
verified the link is working, hence merging the PR

**Pull request merged**